### PR TITLE
Require password confirmation for more plugin operations.

### PR DESCRIPTION
**Pull request by diosmosis**
### Description:

Adding requirement for password confirmation for: activating a plugin, deactivating a plugin, uninstalling a plugin. And for CorePluginsAdmin.setSystemSettings.

### Review

* [ ] Functional review done
* [ ] Potential edge cases thought about (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] Usability review done (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] Security review done [see checklist](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] Code review done
* [ ] Tests were added if useful/possible
* [ ] Reviewed for breaking changes
* [ ] Developer changelog updated if needed
* [ ] Documentation added if needed
* [ ] Existing documentation updated if needed

### CODE REVIEW

**Comment by flamisz** [comment]
Is it common for asking password as a URL parameter when we already have the token validation? Is it necessary? I can't remember using API like this.

**Comment by diosmosis** [comment]
@flamisz if you grep through API files for 'passwordConfirmation', you'll find similar methods. It is an extra security precaution for sensitive settings and changes.

Also just realized I forgot to the UI related changes, will move this back to a draft.

**Comment by diosmosis** [comment]
Added the UI changes. Also noticed we were using the old `ready:` event for materializecss modals, should be using `onOpenEnd` now.

File: plugins/CorePluginsAdmin/Controller.php [hunk 1]
```diff
@@ -425,9 +425,22 @@ public function safemode($lastError = array())
...
     {
-        $pluginName = $this->initPluginModification(static::ACTIVATE_NONCE);
         $this->dieIfPluginsAdminIsDisabled();
 
+        $params = [
+            'module' => 'CorePluginsAdmin',
+            'action' => 'activate',
+            'pluginName' => Common::getRequestVar('pluginName'),
+            'nonce' => Common::getRequestVar('nonce'),
+            'redirectTo' => Common::getRequestVar('redirectTo'),
```
Comment (sgiehl) [inline plugins/CorePluginsAdmin/Controller.php:435]: That actually seems not to work as expected. `redirectTo` might be set to `referer`. In this case the method redirects to the referrer sent with the request. As the password verify now adds another redirect in between, the referrer isn't the original anymore, causing a loop on the confirmation page.

Comment (diosmosis) [reply plugins/CorePluginsAdmin/Controller.php:435]: :+1: should be fixed

**Comment by diosmosis** [comment]
@sgiehl this should be ready for another review

**Review by sgiehl — APPROVED** [review]
Seems to work as expected now. Some UI tests for TagManager and QueuedTracking are failing due to the new password confirmation. Might be good to fix them before merging

**Pull request merged**